core/blockchain.go

```diff
@@ -1250,6 +1262,7 @@ func (bc *BlockChain) writeBlockWithState(block *types.Block, receipts []*types.
 					triedb.Commit(header.Root, true, nil)
 					lastWrite = chosen
 					bc.gcproc = 0
+					bc.sinceFlush = 0
```

Comment:
Do you really need both `sinceFlush` and `lastWrite`? Isn't `sinceFlush` derivable from `lastWrite`?